mappings/net/minecraft/network/MessageType.mapping

```diff
@@ -1,82 +1,140 @@
-		COMMENT Registers the built-in message types and returns the default "chat" type.
 		ARG 0 registry
-	METHOD method_43845 getKey (Ljava/lang/String;)Lnet/minecraft/class_5321;
+	METHOD method_43845 register (Ljava/lang/String;)Lnet/minecraft/class_5321;
 		ARG 0 id
-	CLASS class_7466 NarrationRule
-		COMMENT A narration rule for a message.
-		FIELD comp_795 Ljava/util/Optional;
-			COMMENT the formatting of a message, or {@code Optional.empty()} if the content
-			COMMENT is displayed directly
-		FIELD comp_796 Lnet/minecraft/class_2556$class_7466$class_7467;
-			COMMENT the narration kind, and if this narration can interrupt other ones
+	CLASS class_7466 Narration
+		COMMENT The narration for the message type. This contains the decoration applied
+		COMMENT to the message and the priority of the narration.
 		FIELD field_39233 CODEC Lcom/mojang/serialization/Codec;
-			COMMENT The codec for the message narraton rule.
 		METHOD method_43846 (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/kinds/App;
 			ARG 0 instance
 		METHOD method_43847 of (Lnet/minecraft/class_7463;Lnet/minecraft/class_2556$class_7466$class_7467;)Lnet/minecraft/class_2556$class_7466;
-			COMMENT {@return a rule that narrates a message after formatting} The message
-			COMMENT is narrated as the given {@code kind}, and the narration can include
-			COMMENT sender information in addition to the message content.
+			COMMENT {@return the narration with the specified decoration and priority}
 			ARG 0 decoration
-			ARG 1 kind
+			ARG 1 priority
 		METHOD method_43848 of (Lnet/minecraft/class_2556$class_7466$class_7467;)Lnet/minecraft/class_2556$class_7466;
-			COMMENT {@return a rule that narrates a message's content directly} The message
-			COMMENT is narrated as the given {@code kind}, without any sender-related
-			COMMENT information.
-			ARG 0 kind
-		METHOD method_43849 apply (Lnet/minecraft/class_2561;Lnet/minecraft/class_7436;)Lnet/minecraft/class_2561;
-			COMMENT {@return the text to narrate after applying this rule on the given
-			COMMENT content and sender}
-			ARG 1 content
+			COMMENT {@return the narration with no decoration and the specified priority}
+			COMMENT
+			COMMENT <p>This does not mean the message is not narrated; this means that the message
+			COMMENT is narrated as is without any extra processing.
+			ARG 0 priority
+		METHOD method_43849 applyDecoration (Lnet/minecraft/class_2561;Lnet/minecraft/class_7436;)Lnet/minecraft/class_2561;
+			COMMENT {@return the message with the decoration applied, or {@code message} if there is
+			COMMENT no decoration}
+			ARG 1 message
 			ARG 2 sender
-		CLASS class_7467 Kind
-			COMMENT The narration kind. It also determines if the narration can interrupt
-			COMMENT previous ones.
+		METHOD method_43850 (Lnet/minecraft/class_2561;Lnet/minecraft/class_7436;Lnet/minecraft/class_7463;)Lnet/minecraft/class_2561;
+			ARG 2 decoration
+		CLASS class_7467 Priority
```

Comment:
Revert this? since you've said "kind" makes more sense in NarrationManager context.

Reply:
Maybe. Kind makes more sense, but the field in the JSON is “priority” and I’m not sure what should be changed and what shouldn’t. AKA should the record field be `kind` or `priority`? `Kind priority` doesn’t look great, but `priority` field being named `kind` in the record is confusing too, hmmm

Reply:
@liach Opinions?

Reply:
Having one mismatch in codec serialization is less confusing than inaccurate name being used all over the place imo. It can be clarified in javadoc that the class corresponds to `priority` in json.

Reply:
If we want a compromise, we can consider naming this `PriorityGroup`, `PriorityCategory`, etc. And I suggest rename `hasPriority` to `shouldInterrupt` because `Priority.hasPriority` just makes zero sense.